### PR TITLE
20220121 debugging

### PR DESCRIPTION
**Pull request by mrakitin**
This is a working version of the startup files for AMX that work with a copied conda env from FMX (`collection-2019-3.0-fmx`).

I had to place a placeholder [`.pyOlog.conf`](https://github.com/NSLS-II/profile-collection-ci/blob/main/configs/pyOlog.conf) file to `softioc`'s home dir to enable `%wa` magics.

### CODE REVIEW

File: services/st.cmd [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+export PATH=/opt/conda/bin/:$PATH
+export USER=softioc
+export HOME=/epics/iocs/notebook
+export SHELL=bash
+
+export EPICS_CA_AUTO_ADDR_LIST="no"
+export EPICS_CA_ADDR_LIST="10.66.131.255"
```
Comment (cowanml) [inline services/st.cmd]: suggest replacing these with 1 of (from most general to specific):

```
. /etc/profile
```
```
# relevant snippet from /etc/profile
for i in /etc/profile.d/*.sh /etc/profile.d/sh.local ; do
    if [ -r "$i" ]; then
        if [ "${-#*i}" != "$-" ]; then
            . "$i"
        else
            . "$i" >/dev/null
        fi
    fi
done
```
```
. /etc/profile.d/epics.sh
```
The 1st would seem closest to what you'd get starting a notebook from a terminal.

Comment (cowanml) [reply services/st.cmd]:
```suggestion
. /etc/profile
```

Comment (cowanml) [reply services/st.cmd]: or

Comment (cowanml) [reply services/st.cmd]:
```suggestion
. /etc/profile.d/epics.sh
```

Comment (mrakitin) [reply services/st.cmd]: I like the last option. Thanks, @cowanml!

File: services/st.cmd [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+conda env list
+
+###############################################################################
+# DAMA note (July 10, 2019): the corresponding kernel file is defined in      #
+# /usr/share/jupyter/kernels-available/collection-2019-3.0/kernel.json, and   #
+# symlinked to /usr/share/jupyter/kernels/collection-2019-3.0/kernel.json     #
+###############################################################################
+
+jupyter lab --no-browser --notebook-dir=/epics/iocs/notebook/notebooks/ --ip=0.0.0.0 --port=17000 --NotebookApp.token='' --debug > $jupyter_log 2>&1
```
Comment (mrakitin) [inline services/st.cmd]:
```suggestion
jupyter lab --no-browser --notebook-dir=/epics/iocs/notebook/notebooks/ --ip=0.0.0.0 --port=17000 --debug > $jupyter_log 2>&1
```

File: services/st.cmd [hunk 3]
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+export PATH=/opt/conda/bin/:$PATH
+export USER=softioc
```
Comment (mrakitin) [inline services/st.cmd]:
```suggestion
export USER=xf17id1
```

Comment (JunAishima) [reply services/st.cmd]: is this the script called by `manage-iocs`? is the jupyter-amx.service not used here at all?

Comment (JunAishima) [reply services/st.cmd]: do the ACLs need to be updated for profile_collection to allow the beamline user to modify the files?

other than that, the change seems good

Comment (mrakitin) [reply services/st.cmd]: > is this the script called by `manage-iocs`? is the jupyter-amx.service not used here at all?

The `jupyter-amx.service` file calls this executable. See https://github.com/NSLS-II-AMX/profile_collection/pull/10/files#diff-bf36ad3aec4690ad7ae78e6d23f6d2f539721bfaac0d05ab060f7cd8f3dba5c7R11.

> do the ACLs need to be updated for profile_collection to allow the beamline user to modify the files?
> 
> other than that, the change seems good

I think the `/nsls2/.../profile_collection/` dir is already owned by the xf-user, so we should be good.

File: services/jupyter-amx.service [hunk 4]
```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=AMX Jupyter service
+
+[Service]
+User=softioc
+Group=softioc
```
Comment (mrakitin) [inline services/jupyter-amx.service]: This is done that way at FMX.
```suggestion
User=xf17id1
Group=xf17id1
```

**Comment by JunAishima** [comment]
@danielballan can you review this pull request with a particular eye towards notebook service security? thanks

**Comment by danielballan** [comment]
The suggestion from @mrakitin to remove the token-disabling line should be committed.

File: services/st.cmd (hunk 3 above)
Comment (mrakitin) [reply services/st.cmd]: I think we still need this suggested change to be committed. Thanks!

Comment (JunAishima) [reply services/st.cmd]: I wanted Dan to look over it. he says the same, so committed and resolving.

**Review by JunAishima — APPROVED** [review]

**Review by AbbyGi — APPROVED** [review]

File: services/st.cmd (hunk 3 above)
Comment (mrakitin) [reply services/st.cmd]: I am still seeing `export USER=softioc` in `st.cmd`. We need it to be `xf17id1` as in the .service file above.

**Review by mrakitin — COMMENTED** [review]
Looks good to me. But I can't approve as it's my PR. Feel free to merge.

**Pull request merged**